src/util/actor.js

```diff
@@ -110,40 +110,42 @@ class Actor {
                 cancel();
             }
         } else {
-            // In workers, store the tasks that we need to process before actually processing them. This
-            // is necessary because we want to keep receiving messages, and in particular,
-            // <cancel> messages. Some tasks may take a while in the worker thread, so before
-            // executing the next task in our queue, postMessage preempts this and <cancel>
-            // messages can be processed. We're using a MessageChannel object to get throttle the
-            // process() flow to one at a time.
-            this.tasks[id] = data;
-            this.taskQueue.push(id);
-            if (isWorker()) {
+            if (isWorker() || data.type === 'getResource') {
```

Comment:
Instead of hardcoding the name of the message, can we find another way of marking tasks as immedidate vs. queueable, e.g. by checking for `!data.immediate` here, and adding `immediate as another argument to `Actor#send`?

Reply:
done in d28ac5668f35d9354b9ae1fc7dfbb15b9296aec7